### PR TITLE
Fix authentication issue on API 19

### PR DESCRIPTION
**Pull request by erikghonyan**

### CODE REVIEW

File: auth-sample/src/main/res/layout/activity_main.xml [hunk 1]
```diff
@@ -20,126 +20,69 @@
...
-    android:id="@+id/activity_main"
     xmlns:android="http://schemas.android.com/apk/res/android"
-    xmlns:app="http://schemas.android.com/apk/res-auto"
     xmlns:tools="http://schemas.android.com/tools"
+    android:id="@+id/activity_main"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
     tools:context="com.spotify.sdk.android.authentication.sample.MainActivity">
 
-    <android.support.constraint.ConstraintLayout
```
Comment (d-kunin) [inline auth-sample/src/main/res/layout/activity_main.xml]: What was the change here? Does not seem related to the initial problem at the first sight.

File: auth-lib/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -44,7 +44,6 @@
         </activity>
 
         <activity
-            android:launchMode="singleInstance"
```
Comment (d-kunin) [inline auth-lib/src/main/AndroidManifest.xml]: I am not sure why it was set before, but I am pretty sure it was to fix some other bug. 
Let live without it for now and please run an extensive manual testing session trying to login from different activities and other weird cases.

File: auth-sample/src/main/res/layout/activity_main.xml (hunk 1 above)
Comment (erikghonyan) [inline auth-sample/src/main/res/layout/activity_main.xml]: There was a crash related to ellipsizing the text, which could be fixed by removing that attribute, but then the layout would break (views would overlap). Since this is just a demo I decided to replace this with a simple LinearLayout instead of trying to fix the constraints on this layout.
More info in the commit message.

File: auth-lib/src/main/AndroidManifest.xml (hunk 2 above)
Comment (erikghonyan) [inline auth-lib/src/main/AndroidManifest.xml]: It was added in 8a97adf , I assumed we needed this so that we don't launch multiple instances of this activity.

Comment (erikghonyan) [inline auth-lib/src/main/AndroidManifest.xml]: I removed FLAG_ACTIVITY_SINGLE_TOP from createLoginActivityIntent because it prevents the system from re-creating the LoginActivity and calls onNewIntent instead, which we use to handle web redirects (and return the token/code).

This happens when you call openLoginActivity when LoginActivity is visible

**Comment by corriejgreen** [comment]
Just came across this issue, I cloned this 3 days ago so shouldnt have the issue? Can anyone share the solution so I can debug?

**Comment by turing-tech** [comment]
Also still encountering this issue.
